src/rust/rust-proto-new/src/graplinc/grapl/api/pipeline_ingress/v1beta1.rs

```diff
@@ -0,0 +1,143 @@
+        PublishRawLogsRequestProto {
+            event_source_id: Some(publish_raw_logs_request.event_source_id.into()),
+            tenant_id: Some(publish_raw_logs_request.tenant_id.into()),
+            log_event: publish_raw_logs_request.log_event.to_vec(),
+        }
+    }
+}
+
+impl type_url::TypeUrl for PublishRawLogsRequest {
+    const TYPE_URL: &'static str =
+        "graplsecurity.com/graplinc.grapl.api.pipeline_ingress.v1beta1.PublishRawLogsRequest";
+}
+
+impl SerDe for PublishRawLogsRequest {
+    fn serialize(self) -> Result<Bytes, SerDeError> {
+        let publish_raw_logs_request_proto = PublishRawLogsRequestProto::from(self);
+        let mut buf = BytesMut::with_capacity(publish_raw_logs_request_proto.encoded_len());
+        publish_raw_logs_request_proto.encode(&mut buf)?;
+        Ok(buf.freeze())
+    }
+
+    fn deserialize<B>(buf: B) -> Result<Self, SerDeError>
+    where
+        B: bytes::Buf,
+        Self: Sized,
+    {
+        let publish_raw_logs_request_proto: PublishRawLogsRequestProto = Message::decode(buf)?;
+        publish_raw_logs_request_proto.try_into()
+    }
+}
+
+//
+// PublishRawLogsResponse
+//
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct PublishRawLogsResponse {
+    pub created_time: SystemTime,
+}
+
+impl TryFrom<PublishRawLogsResponseProto> for PublishRawLogsResponse {
+    type Error = SerDeError;
+
+    fn try_from(
+        publish_raw_logs_response_proto: PublishRawLogsResponseProto,
+    ) -> Result<Self, Self::Error> {
+        let created_time = publish_raw_logs_response_proto
+            .created_time
+            .ok_or_else(|| SerDeError::MissingField("created_time".to_string()));
+
+        Ok(PublishRawLogsResponse {
+            created_time: created_time?.try_into()?,
+        })
+    }
+}
+
+impl TryFrom<PublishRawLogsResponse> for PublishRawLogsResponseProto {
+    type Error = SerDeError;
+
+    fn try_from(publish_raw_logs_response: PublishRawLogsResponse) -> Result<Self, Self::Error> {
+        Ok(PublishRawLogsResponseProto {
+            created_time: Some(publish_raw_logs_response.created_time.try_into()?),
+        })
+    }
+}
+
+impl type_url::TypeUrl for PublishRawLogsResponse {
+    const TYPE_URL: &'static str =
+        "grapsecurity.com/graplinc.grapl.api.pipeline_ingress.v1beta1.PublishRawLogsResponse";
+}
+
+impl SerDe for PublishRawLogsResponse {
+    fn serialize(self) -> Result<Bytes, SerDeError> {
+        let publish_raw_logs_response_proto = PublishRawLogsResponseProto::try_from(self)?;
+        let mut buf = BytesMut::with_capacity(publish_raw_logs_response_proto.encoded_len());
+        publish_raw_logs_response_proto.encode(&mut buf)?;
+        Ok(buf.freeze())
+    }
+
+    fn deserialize<B>(buf: B) -> Result<Self, SerDeError>
+    where
+        B: bytes::Buf,
+        Self: Sized,
+    {
+        let publish_raw_logs_response_proto: PublishRawLogsResponseProto = Message::decode(buf)?;
+        publish_raw_logs_response_proto.try_into()
```

Comment:
🤔 It might make things a bit more concise to just call this variable `response_proto`, since the type hint and the identity of `self` in this `impl` block make it clear that it's for the PublishRawLogs part of the API. This applies (modulo the names, of course) to the other serialize/deserialize implementations in this file.

I don't think the shorter name sacrifices any meaning or context, and makes the code a bit easier to read (IMO, at least).

(This is just a suggestion you can take or leave as you like.)

Reply:
fixed